src/mono/sample/iOS-NativeAOT/Program.csproj

```diff
@@ -0,0 +1,96 @@
+    <OutputPath>bin</OutputPath>
+    <IntermediateOutputPath>$(MSBuildThisFileDirectory)/obj/</IntermediateOutputPath>
+    <TargetFramework>net7.0</TargetFramework>
+    <TargetOS Condition="'$(TargetOS)' == ''">ios</TargetOS>
+    <TargetOS Condition="'$(TargetsiOSSimulator)' == 'true'">iossimulator</TargetOS>
+    <DeployAndRun Condition="'$(DeployAndRun)' == ''">true</DeployAndRun>
+    <RuntimeIdentifier>$(TargetOS)-$(TargetArchitecture)</RuntimeIdentifier>
+    <AppName>HelloiOS</AppName>
+    <StripDebugSymbols Condition="'$(StripDebugSymbols)' == ''">false</StripDebugSymbols>
+    <!-- NativeAOT-specific props -->
+    <NativeLib>static</NativeLib>
+    <CustomNativeMain>true</CustomNativeMain>
+    <UseNativeAOTRuntime Condition="'$(UseNativeAOTRuntime)' == ''">true</UseNativeAOTRuntime>
+    <!-- FIXME: Once we support building System.Globalization.Native and icu, should be removed -->
+    <InvariantGlobalization>true</InvariantGlobalization>
+    <!-- FIXME: We do not use publish targets yet, but we need to create a publish directory -->
+    <PublishDir Condition="'$(PublishDir)' == ''">$(OutputPath)/publish</PublishDir>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <Compile Include="..\iOS\Program.cs" Link="Program.cs" />
+    <DirectPInvoke Include="__Internal" />
+  </ItemGroup>
+
+  <PropertyGroup Condition="'$(TargetOS)' == 'maccatalyst'">
+    <DevTeamProvisioning Condition="'$(TargetOS)' == 'maccatalyst' and '$(DevTeamProvisioning)' == ''">adhoc</DevTeamProvisioning>
+    <EnableAppSandbox Condition="'$(EnableAppSandbox)' == ''">false</EnableAppSandbox>
+  </PropertyGroup>
+
+  <Import Project="$(MSBuildThisFileDirectory)../../../coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets" />
+  <UsingTask TaskName="AppleAppBuilderTask"
+             AssemblyFile="$(AppleAppBuilderTasksAssemblyPath)" />
+
+  <!-- FIXME: Once we set up builing appropriate runtime package for iOS-like platforms the following properties should be removed -->
+  <PropertyGroup>
+      <NativeAotFrameworkLibsDir>$(RepoDir)/artifacts/bin/runtime/net8.0-$(TargetOS)-$(Configuration)-$(TargetArchitecture)/</NativeAotFrameworkLibsDir>
+      <NativeAotRuntimeLibsDir>$(RepoDir)/artifacts/bin/coreclr/$(TargetOS).$(TargetArchitecture).$(Configuration)/</NativeAotRuntimeLibsDir>
+      <IlcHostArchitecture>$([System.Runtime.InteropServices.RuntimeInformation]::OSArchitecture.ToString().ToLowerInvariant)</IlcHostArchitecture>
+      <IlcPath>$(RepoDir)/artifacts/bin/coreclr/osx.$(IlcHostArchitecture).$(Configuration)/ilc/</IlcPath>
+      <IlcToolsPath>$(IlcPath)</IlcToolsPath>
+      <IlcSdkPath>$(NativeAotRuntimeLibsDir)/aotsdk/</IlcSdkPath>
+      <IlcFrameworkPath>$(NativeAotFrameworkLibsDir)</IlcFrameworkPath>
+      <IlcFrameworkNativePath>$(NativeAotFrameworkLibsDir)</IlcFrameworkNativePath>
+  </PropertyGroup>
+
+  <Target Name="BuildAppBundle"
+    AfterTargets="Build"
+    DependsOnTargets="SetupProperties;ComputeIlcCompileInputs;IlcCompile">
```

Comment:
Nit: I've searched for a target that depends on them, like "publish app" but it seems that there is no such target. Not sure if it is a full list of targets needed before AppleAppBuilderTask.

Reply:
For publishing with NativeAOT it is expected to set `PublishAOT=true` which would make the SDK include NativeAOT build integration targets: https://github.com/dotnet/sdk/blob/7ae63298117b56fedca0985ac712113354491756/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets#L1189 
However, since this requires the full support for iOS with NativeAOT (runtime packs, SDK integration, etc) we cannot use the publish targets. For that reason, we just cherry-pick the targets we need to gather required dependencies and compile the application with `ilc`.